core/src/main/scala/kafka/message/MessageSet.scala

```diff
@@ -70,7 +90,12 @@ abstract class MessageSet extends Iterable[MessageAndOffset] {
     * Less than the complete amount may be written, but no more than maxSize can be. The number
     * of bytes written is returned */
   def writeTo(channel: GatheringByteChannel, offset: Long, maxSize: Int): Int
-  
+
+  /**
+   * Check if all the messages in the message set have the expected magic value
+   */
+  def magicValueInAllMessages(expectedMagicValue: Byte): Boolean
```

Comment:
The method only iterates shallow messages. So, perhaps changing the method to magicValueInAllWrapperMessages and adjusting the comments?